internal/handlers/pg/msg_update.go

```diff
@@ -67,18 +66,17 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		return nil, err
 	}
 
-	created, err := pgdb.CreateCollectionIfNotExist(ctx, h.PgPool, sp.DB, sp.Collection)
+	err = h.PgPool.InTransactionRetry(ctx, func(tx pgx.Tx) error {
+		return pgdb.CreateCollectionIfNotExists(ctx, tx, sp.DB, sp.Collection)
+	})
 	if err != nil {
-		if errors.Is(pgdb.ErrInvalidTableName, err) ||
+		if errors.Is(pgdb.ErrInvalidCollectionName, err) ||
```

Comment:
It's not really possible to test it in FerretDB because we have custom rules for collection names (different than what MongoDB has), see how `ErrInvalidCollectionName` is used in `pgdb`.

Reply:
Or maybe we can add a case that could be invalid for both MongoDB and us. I think we can even add it to the list of collections when setup tests. I'll create an issue.

Reply:
https://github.com/FerretDB/FerretDB/issues/1688